mindsdb/integrations/handlers/llama_index_handler/llama_index_handler.py

```diff
@@ -0,0 +1,165 @@
+  
+    def create(self, target: str, df: Optional[pd.DataFrame] = None, args: Optional[Dict] = None) -> None:
+        
+        
+        if 'using' not in args:
+            raise Exception("LlamaIndex engine requires a USING clause! Refer to its documentation for more details.")
+
+        if args['using']['reader'] == 'SimpleWebPageReader':
+            if 'source_url_link' not in args['using']:
+                raise Exception("LlamaIndex engine requires a source_url_link parameter.Refer to its documentation for more details.")
+
+        if 'openai_api_key' not in args['using']:
+            raise Exception("LlamaIndex engine requires a openai_api_key parameter.Refer to its documentation for more details.")
+
+        if 'index_class' not in args['using']:
+            args['using']['index_class'] = self.default_index_class
+        elif args['using']['index_class'] not in self.supported_index_class:
+            raise Exception(f"Invalid index class argument. Please use one of {self.supported_index_class}")
+        
+        if 'query_engine' not in args['using']:
+            args['using']['query_engine'] = self.default_query_engine
+        elif args['using']['query_engine'] not in self.supported_query_engine:
+            raise Exception(f"Invalid operation mode. Please use one of {self.supported_query_engine}")
+
+        if 'reader' not in args['using']:
+            args['using']['reader'] = self.default_reader
+        elif args['using']['reader'] not in self.supported_reader:
+            raise Exception(f"Invalid operation mode. Please use one of {self.supported_query_engine}")
+
+       
+        documents_df_reader = []
+        documents_url_reader = []
+        pred = None
+        if args['using']['reader'] == 'DFReader':
+            for row in df.itertuples():
+                doc_str = ", ".join([str(entry) for entry in row])
+                documents_df_reader.append(Document(doc_str))  
+            pred = documents_df_reader   
+      
+        elif args['using']['reader'] == 'SimpleWebPageReader':
+            SimpleWebPageReader = download_loader("SimpleWebPageReader")
+            documents_url_reader = SimpleWebPageReader(html_to_text=True).load_data([args['using']['source_url_link']])
+            pred = documents_url_reader 
+    
+        
+        self.model_storage.file_set('pred', dill.dumps(pred))
+        self.model_storage.json_set('args', args)
+
+        
+
+    def predict(self, df: Optional[pd.DataFrame] = None, args: Optional[Dict] = None) -> None:
+
+        args = self.model_storage.json_get('args')
+        data_docs = dill.loads(self.model_storage.file_get('pred'))
+        input_keys = list(args.keys())
+
+        input_column = args['using']['input_column']
+
+        if input_column not in df.columns:
+            raise RuntimeError(f'Column "{input_column}" not found in input data')
+
+        if args['using']['reader'] == 'DFReader':  
```

Comment:
With the suggested exception inside `create`, these two `if`s wouldn't be needed, just define `query_engine`.

Reply:
Thanks for suggestion @paxcema, I have made the suggested changes